File: src/core/redis.py
import redis.asyncio as redis
from src.core.config import settings
from loguru import logger

class RedisClient:
    def __init__(self):
        self.url = settings.REDIS_URL
        self._client: redis.Redis = None

    async def connect(self):
        """Estabiliza a conexão assíncrona com o Redis."""
        try:
            self._client = redis.from_url(
                self.url, 
                encoding="utf-8", 
                decode_responses=True
            )
            await self._client.ping()
            logger.info(f"🔌 Conectado ao Redis em {self.url}")
        except Exception as e:
            logger.error(f"❌ Erro ao conectar no Redis: {e}")
            raise

    async def disconnect(self):
        """Fecha a conexão com o Redis."""
        if self._client:
            await self._client.close()
            logger.info("🔌 Conexão Redis encerrada.")

    async def get(self, key: str) -> str:
        return await self._client.get(key)

    async def set(self, key: str, value: str, expire: int = None):
        await self._client.set(key, value, ex=expire)

    async def delete(self, key: str):
        await self._client.delete(key)

    async def exists(self, key: str) -> bool:
        return await self._client.exists(key) > 0

redis_client = RedisClient()

File: node-version/src/config/redis.js
const Redis = require('ioredis');
const logger = require('../utils/logger');
require('dotenv').config({ path: '../.env' });

/**
 * 3. Configuração do Redis (Substitui redis-py assíncrono)
 * Responsável pelo controle de presença e mensageria distribuída.
 */
class RedisClient {
  constructor() {
    const redisUrl = process.env.REDIS_URL || 'redis://127.0.0.1:6379/0';
    this.client = new Redis(redisUrl, {
      maxRetriesPerRequest: null, // Útil para workers pesados e queues
    });

    this.client.on('connect', () => {
      logger.info(`🔌 Conectado ao Redis em ${redisUrl}`);
    });

    this.client.on('error', (err) => {
      logger.error(`❌ Erro no Redis: ${err.message}`);
    });
  }

  async set(key, value, expireSeconds = null) {
    if (expireSeconds) {
      await this.client.set(key, value, 'EX', expireSeconds);
    } else {
      await this.client.set(key, value);
    }
  }

  async get(key) {
    return await this.client.get(key);
  }

  async delete(key) {
    await this.client.del(key);
  }

  async disconnect() {
    await this.client.quit();
    logger.info('🛑 Conexão com Redis encerrada.');
  }
}

const redisService = new RedisClient();
module.exports = redisService;
